File: 🟡 Medium/2419 - Longest Subarray With Maximum Bitwise AND/_2419LongestSubarrayWithMaximumBitwiseAND.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function (nums) {
    let maxi = 0;
    let currLongestSeq = 0;
    let res = 0;

    for (let i = 0; i < nums.length; i++) {
        if (maxi < nums[i]) {
            maxi = nums[i];
            currLongestSeq = 0;
            res = 0;
        }

        if (maxi == nums[i]) {
            currLongestSeq++;
        } else {
            currLongestSeq = 0;
        }

        res = Math.max(res, currLongestSeq);
    }
    return res;
};

File: 🟡 Medium/2419 - Longest Subarray With Maximum Bitwise AND/_2419LongestSubarrayWithMaximumBitwiseAND.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maxi = 0;
        int currMax = 0;
        int res = 0;

        for (int i = 0; i < nums.length; i++) {
            if (maxi < nums[i]) {
                maxi = nums[i];
                currMax = 0;
                res = 0;
            }

            if (maxi == nums[i]) {
                currMax++;
            } else {
                currMax = 0;
            }

            res = Math.max(res, currMax);
        }
        return res;
    }
}